web-app/prisma/schema.prisma

```diff
@@ -353,94 +354,85 @@ enum AgentListType {
 }
 
 model CreditTransaction {
-  id            String                @id @default(cuid())
-  createdAt     DateTime              @default(now())
-  updatedAt     DateTime              @updatedAt
-  amount        BigInt //positive is always a credit topup and negative is always a credit spend
-  includedFee   BigInt
-  note          String?
-  noteKey       String?
-  errorNote     String?
-  errorNoteKey  String?
-  type          CreditTransactionType
-  userId        String
-  user          User                  @relation(fields: [userId], references: [id], onDelete: Cascade)
-  fiatPayment   FiatPayment?          @relation(fields: [fiatPaymentId], references: [id])
-  fiatPaymentId String?               @unique
-
-  status      CreditTransactionStatus @default(PENDING)
-  job         Job?                    @relation("JobCost")
-  refundedJob Job?                    @relation("JobRefund")
-}
-
-enum CreditTransactionStatus {
-  PENDING
-  SUCCEEDED
-  FAILED
-}
+  id        String   @id @default(cuid())
+  createdAt DateTime @default(now())
+  updatedAt DateTime @updatedAt
 
-enum CreditTransactionType {
-  TOP_UP
```

Comment:
Why did we remove the type?

If it is implicit, then we at least need docs on how it works. But I would like it explicit, to catch possible errors

Reply:
What do we need this type for right now? The negative and positive amounts tell us what we need to know, don't they? Additionally the connection to a `Job` or a `FiatTransaction` has additional information. The type doesn't belong to the database for me, it could become a computed property which is added based on the data.

Reply:
It is implicit and only know if you combine some internal knowledge e.g. if job is set and amount is negative -> refund.

I think an explicit state is useful to catch potential problems

Reply:
How does it help us if the underlying data doesn't fit to the type? What's wrong the type or the data? The data is the source of truth. Can we make sure that the type is automatically derived from the data within prisma?

Reply:
If we want/need it, let's create an issue and implement it using [client-extensions](https://www.prisma.io/docs/orm/prisma-client/client-extensions/result) on `CreditTransaction`. This approach will derive the type from the data.

What do you think?

Reply:
I implement a very simple variant with b30245f721b59de20ec006cd8d53f43a6fc90222

This way, the type is derived from the data; otherwise, the type does not provide reliable information. We can enhance this implementation as needed later on.

Reply:
The Idee was to spot of there Are mismatches or data coruptions unauthorized Change more Daily